### ISSUE TITLE
Back Button to NS app crashes the app.

### ISSUE THREAD

**Issue by sitefinitysteve** [NativeScript/NativeScript#2042]
### Did you verify this is a real problem by searching
Yes

### Tell us about the problem
Please, ensure your title is less than 63 characters long and starts with a capital
letter.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.0
- Cross-platform modules: (check the 'version' attribute in the
`node_modules/tns-core-modules/package.json` file in your project)
- Runtime(s): (look for the `"tns-android"` and `"tns-ios"` properties in the
`package.json` file of your project)
- Plugin(s): 
    "nativescript-appversion": "^1.1.3",
    "nativescript-dom": "^1.0.0",
    "nativescript-effects": "^0.2.3",
    "nativescript-floatingactionbutton": "^0.2.5",
    "nativescript-google-analytics": "^0.2.4",
    "nativescript-snackbar": "file:../../../plugins/snackbar/nativescript-snackbar",
    "nativescript-social-share": "^1.1.1",
    "nativescript-unit-test-runner": "^0.2.8",
    "nativescript-zendesk": "^0.2.8",
    "tns-core-modules": "^2.0.0"

### Please tell us how to recreate the issue in as much detail as possible.
```
var util = require("utils/utils");
util.openUrl("https://www.google.com");
```
So this opens google through your app, click the android back button, and when the app loads back I get hit with this
```
java.lang.ClassCastException: org.nativescript.widgets.CommonLayoutParams cannot be cast to android.widget.AbsListView$LayoutParams
at android.widget.AbsListView$RecycleBin.fillActiveViews(AbsListView.java:8793)
at android.widget.ListView.layoutChildren(ListView.java:1684)
at android.widget.AbsListView.onLayout(AbsListView.java:2723)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at org.nativescript.widgets.CommonLayoutParams.layoutChild(CommonLayoutParams.java:189)
at org.nativescript.widgets.GridLayout.onLayout(GridLayout.java:353)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at android.support.v4.view.ViewPager.onLayout(ViewPager.java:1695)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at org.nativescript.widgets.CommonLayoutParams.layoutChild(CommonLayoutParams.java:189)
at org.nativescript.widgets.GridLayout.onLayout(GridLayout.java:353)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at org.nativescript.widgets.CommonLayoutParams.layoutChild(CommonLayoutParams.java:189)
at org.nativescript.widgets.GridLayout.onLayout(GridLayout.java:353)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at org.nativescript.widgets.CommonLayoutParams.layoutChild(CommonLayoutParams.java:189)
at org.nativescript.widgets.ContentLayout.onLayout(ContentLayout.java:73)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:344)
at android.widget.FrameLayout.onLayout(FrameLayout.java:281)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1742)
at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at android.widget.FrameLayout.layoutChildren(FrameLayout.java:344)
at android.widget.FrameLayout.onLayout(FrameLayout.java:281)
at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:3193)
at android.view.View.layout(View.java:17938)
at android.view.ViewGroup.layout(ViewGroup.java:5812)
at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2666)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2367)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1437)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7397)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:920)
at android.view.Choreographer.doCallbacks(Choreographer.java:695)
at android.view.Choreographer.doFrame(Choreographer.java:631)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:906)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:158)
at android.app.ActivityThread.main(ActivityThread.java:7229)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
Project:
https://www.dropbox.com/s/on0tttujo04g0pl/OutbreakHelp_20.zip?dl=0
Open Resources, click a resource, then click the download button
When the page opens, click the android back button

**Comment by tsonevn** [comment]
Hi @sitefinitysteve, 
Thank you for your issue.
I reviewed your project, actually it works correct and I was not able to reproduce this problem.
Could you reproduce this issue in a sample app without plugins. In addition, could you check, whether the `tns-module` version in `platform/android/src/main/assets/app/tns-modules/package.json` is `2.0`. Sometime the version is not update correctly.
I hope this information would be useful. I will be glad to assist you further.

**Comment by sitefinitysteve** [comment]
1000% on 2.0 :/
```
  "name": "tns-core-modules",
  "description": "Telerik NativeScript Core Modules",
  "version": "2.0.0",
```

The view is in /views/resources/detail

Maybe you're not clicking what I think you are?
Video: http://www.screencast.com/t/g6jedY0rtm

Console shows this
```
JS: Binding: Property: 'Title' is invalid or does not exist. SourceProperty: 'data.Title'
JS: Binding: Property: 'ArbitraryDateString' is invalid or does not exist. SourceProperty: 'data.ArbitraryDateString'
JS: Binding: Property: 'ExecutiveSummary' is invalid or does not exist. SourceProperty: 'data.ExecutiveSummary'
JS: Binding: Property: 'Content' is invalid or does not exist. SourceProperty: 'data.Content'
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Property: 'Type' is invalid or does not exist. SourceProperty: 'data.Type.Text'
JS: Binding: Property: 'Type' is invalid or does not exist. SourceProperty: 'data.Type.Text'
JS: Binding: Property: 'Author' is invalid or does not exist. SourceProperty: 'data.Author'
JS: Binding: Property: 'LevelOfEvidenceString' is invalid or does not exist. SourceProperty: 'data.LevelOfEvidenceString'
JS: Binding: Property: 'Categories' is invalid or does not exist. SourceProperty: 'data.Categories'
JS: Binding: Property: 'Jurisdictions' is invalid or does not exist. SourceProperty: 'data.Jurisdictions'
JS: Binding: Property: 'RelatedResources' is invalid or does not exist. SourceProperty: 'data.RelatedResources'
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
JS: Binding: Run-time error occured in file: undefined at line: undefined and column: undefined
```

**Comment by vjoao** [comment]
Sorry, cannot even build your project on Windows because
`C:\Users\Victor\Desktop\OutbreakHelp\platforms\android\build\intermediates\res\merged\nativescriptfloatingactionbuttonNativescript-google-analyticsNativescript-snackbarNativescript-zendesk\debug\drawable-ldrtl-hdpi-v17\abc_ic_ab_back_mtrl_am_alpha.png`
is a path with more than 240 characters.

Unrelated issue, but why the CLI merges all the plugin names? Kinda sucks.

**Comment by vjoao** [comment]
@sitefinitysteve I can reproduce the issue on a real device as well, running android 5.1.1.

**Comment by sitefinitysteve** [comment]
I very very much appreciate you testing it. I can repro on physical 6,and all geny versions as well :/

**Comment by vjoao** [comment]
Another path that causes the bug but now it's not even coming back from the browser:

Webinars -> click first item -> view -> (when page load, click back button) -> app page shows just fine. -> press back button again: same crash.

**Comment by sitefinitysteve** [comment]
@tsonevn Should be reproducible... I'm even using osx and @vjoao is on windows... can we get this into the next hotfix, she's kind of a blocking bug on release.

**Comment by tsonevn** [comment]
Hi @sitefinitysteve,

I was able to reproduce the abnormality. Unfortunately I confirm that this is a real issue.
We will do everything possible to solve the problem. 

Regards,
@tsonevn

**Comment by sitefinitysteve** [comment]
@tsonevn Thanks ever so much... I was given the go-head to publish the app officially, but this now kinda prevents that from happening.

**Comment by vjoao** [comment]
@sitefinitysteve have you thought about using WebViews like in-app browsers to display the content?

**Comment by hshristov** [comment]
@sitefinitysteve @vjoao This should be fixed with 2.1.0
